src/client/sandbox/node/window.js

```diff
@@ -837,5 +838,65 @@ export default class WindowSandbox extends SandboxBase {
                 return nativeMethods.DOMParserParseFromString.apply(this, args);
             };
         }
+
+        overrideDescriptor(window.Node.prototype, 'firstChild', {
+            getter: function () {
+                if (ShadowUI.isShadowContainer(this))
+                    return windowSandbox.shadowUI.getFirstChild(this);
+
+                return nativeMethods.nodeFirstChildGetter.call(this);
+            }
+        });
+
+        overrideDescriptor(window.Element.prototype, 'firstElementChild', {
+            getter: function () {
+                if (ShadowUI.isShadowContainer(this))
+                    return windowSandbox.shadowUI.getFirstElementChild(this);
+
+                return nativeMethods.elementFirstElementChildGetter.call(this);
+            }
+        });
+
+        overrideDescriptor(window.Node.prototype, 'lastChild', {
+            getter: function () {
+                if (ShadowUI.isShadowContainer(this))
+                    return windowSandbox.shadowUI.getLastChild(this);
+
+                return nativeMethods.nodeLastChildGetter.call(this);
+            }
+        });
+
+        overrideDescriptor(window.Element.prototype, 'lastElementChild', {
+            getter: function () {
+                if (ShadowUI.isShadowContainer(this))
+                    return windowSandbox.shadowUI.getLastElementChild(this);
+
+                return nativeMethods.elementLastElementChildGetter.call(this);
+            }
+        });
+
+        overrideDescriptor(window.Node.prototype, 'nextSibling', {
+            getter: function () {
+                let el = this;
```

Comment:
Move this code to the `shadowUI.getNextSibling(el)` function
and use it for `nextSibling` and `nextElementSibling` properties